source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_property.rst

```diff
@@ -39,12 +40,20 @@ Matrix properties
           the matrix is square, the user data contain both lower and upper
           triangular regions, and that its transpose is equal to itself.
       * - ``sorted``
-        - | Guarantees that the user-provided matrix data has some sorting
-            property.
+        - | Guarantees that the user-provided matrix data is fully sorted.
           | For CSR this guarantees that the column indices are sorted in
             ascending order for a given row.
           | For COO this guarantees that the indices are sorted by row then by
             column in ascending order.
+      * - ``sorted_by_rows``
+        - | Guarantees that the user-provided matrix data is partially sorted.
+            Has no effect if the property ``sorted`` is set.
+          | For COO this guarantees that the indices are sorted by row in
+            ascending order. The column indices within a row do not need to be
+            sorted.
+          | Setting this property for a matrix using the CSR format will throw an
+            :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+            exception.
```

Comment:
Why should it throw an `invalid_argument` exception for CSR format? This is the default state of a CSR format anyway. Setting this property on a CSR matrix should do nothing regardless of whether the `sorted` property has been set or not.

Reply:
I don't really think of the CSR as being sorted by rows in the sense that the values in `row_ptr` don't represent indices and can't be ordered in any other way. I understand the idea for allowing this. I think not throwing an exception is a risk for inexperienced users that maybe wanted to set the `sorted` property instead.

Reply:
> I think not throwing an exception is a risk for inexperienced users that maybe wanted to set the sorted property instead.

Good thinking. This is fine then.